packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -94,8 +94,10 @@ export class DiscoveryWatcher {
     )
 
     if (diff.length > 0) {
-      const message = diffToMessage(name, diff)
-      await this.notify(message)
+      const messages = diffToMessage(name, diff)
+      for (const message of messages) {
```

Comment:
Maybe instead of calling notify multiple times, we also accept string[] as an argument of notify and handle message arrays inside the function